src/ray/gcs/tables.h

```diff
@@ -81,20 +93,25 @@ class Table {
   Status Subscribe(const JobID &job_id,
                    const ID &id,
                    const Callback &subscribe,
-                   const Callback &done);
+                   const Callback &done) {
+    return Status::NotImplemented("Table::Subscribe is not implemented");
+  }
 
   /// Remove and entry from the table
   Status Remove(const JobID &job_id, const ID &id, const Callback &done);
 
- private:
+ protected:
   std::unordered_map<ID, std::unique_ptr<CallbackData>, UniqueIDHasher>
       callback_data_;
   std::shared_ptr<RedisContext> context_;
+  AsyncGcsClient *client_;
```

Comment:
why `protected`? for tests?

Reply:
so inherited classes like TaskTable can access it